### PR TITLE
better tooling around fixing up cases with bad index references

### PR DESCRIPTION
**Pull request by czue**
if a referenced case actually exists just syncs it to elasticsearch
if a referenced case is deleted, delete its entire dependent network (interactively with warnings)
if a referenced case is not found, delete the referring case and its network (interactively with warnings)

have successfully run this on enikshay-test and resolved all issues.

http://manage.dimagi.com/default.asp?243839

@emord cc @NoahCarnahan buddy

### CODE REVIEW

File: corehq/apps/hqadmin/management/commands/check_case_index_ids.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
 import csv
-from django.core.management import BaseCommand
-from casexml.apps.case.cleanup import close_cases
+from django.core.management import BaseCommand, call_command
+from casexml.apps.case.models import CommCareCase
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/check_case_index_ids.py]: F401 'CommCareCase' imported but unused

File: corehq/apps/hqadmin/management/commands/check_case_index_ids.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 from corehq.apps.receiverwrapper.util import get_app_version_info
-from corehq.apps.reports.views import FormDataView
+from corehq.apps.reports.util import resync_case_to_es
 from corehq.apps.users.util import cached_owner_id_to_display
 from corehq.elastic import ES_MAX_CLAUSE_COUNT
 from corehq.apps.es.cases import CaseES
-from corehq.form_processor.interfaces.dbaccessors import FormAccessors
-from corehq.util.view_utils import absolute_reverse
+from corehq.form_processor.change_publishers import publish_case_saved
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/check_case_index_ids.py]: F401 'publish_case_saved' imported but unused

File: corehq/apps/hqadmin/management/commands/check_case_index_ids.py [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors, CaseAccessors
+from corehq.form_processor.utils import should_use_sql_backend
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/check_case_index_ids.py]: F401 'should_use_sql_backend' imported but unused

**Review by NoahCarnahan — APPROVED** [review]
👍

**Pull request merged**